serverless-map-reduce/amr.js

```diff
@@ -0,0 +1,61 @@
+const invoke = (() => {
+  const request = require('request-promise-native');
+
+  return async (fname, fargs) => {
+    console.log(`Invoking ${fname}()`);
+    const uri = `https://run.binaris.com/v2/run/${process.env.BINARIS_ACCOUNT_ID}/${fname}`;
+    const res = await request.post(uri, {
+      json: true,
+      body: fargs
+    });
+    console.log(`${fname}() -> ${res}`);
+    return res;
+  };
+})();
+
+const getRedisConnection = (() => {
+  const Redis = require('ioredis');
+  const { REDIS_HOST, REDIS_PORT, REDIS_PASSWORD } = process.env;
+
+  const connectToRedis = () => {
+    console.log(`Connecting to Redis at ${REDIS_HOST}:${REDIS_PORT}`);
+    return new Redis({
+      host: REDIS_HOST,
+      port: REDIS_PORT,
+      password: REDIS_PASSWORD,
+      tls: { rejectUnauthorized: false },
+    });
+  };
+
+  let redis;
+  return () => redis || (redis = connectToRedis());
+})();
+
+exports.controller = async job => {
+  job.uid = `job${Math.random().toString(36).substring(2, 15)}${Math.random().toString(36).substring(2, 15)}`;
```

Comment:
Please put a comment explaining what this does, as of now it's actually Disney tier magic.

Reply:
Or just change it to be obvious.

Reply:
https://www.npmjs.com/package/uuidv4